dialer.go

```diff
@@ -153,16 +165,81 @@ func WithConnectionProperties(props amqp.Table) Option {
 	}
 }
 
+func WithReadyChan(receiver chan struct{}) Option {
+	return func(c *Dialer) {
+
+		if receiver == nil {
+			receiver = make(chan struct{}, 1)
+		} else {
+			if cap(receiver) == 0 {
+				panic(errChanIsUnbuffered)
+			}
+		}
+
+		c.notifyReady = append(c.notifyReady, receiver)
+	}
+}
+
+func WithUnreadyChan(receiver chan error) Option {
+	f := func(c *Dialer) {
+
+		if receiver == nil {
+			receiver = make(chan error, 1)
+		} else {
+			if cap(receiver) == 0 {
+				panic(errChanIsUnbuffered)
+			}
+		}
+
+		c.notifyUnready = append(c.notifyUnready, receiver)
+	}
+	return f
+}
+
 func (c *Dialer) ConnectionCh() <-chan *Connection {
 	return c.connCh
 }
 
-func (c *Dialer) NotifyReady() <-chan struct{} {
-	return c.readyCh
+func (c *Dialer) NotifyReady(receiver chan struct{}) <-chan struct{} {
+	if cap(receiver) == 0 {
+		panic(errChanIsUnbuffered)
+	}
+
+	select {
+	case <- c.NotifyClosed():
+		return receiver
+	default:
+		c.mu.Lock()
```

Comment:
do `defer c.mu.Unlock()` as the next line after lock

Reply:
fixed